src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs

```diff
@@ -965,6 +977,11 @@ public SyntaxNodeAnalysisContext(SyntaxNode node, SemanticModel semanticModel, A
             _cancellationToken = cancellationToken;
         }
 
+        public SyntaxNodeAnalysisContext(SyntaxNode node, SemanticModel semanticModel, AnalyzerOptions options, Action<Diagnostic> reportDiagnostic, Func<Diagnostic, bool> isSupportedDiagnostic, CancellationToken cancellationToken)
```

Comment:
Can we instead make the new API just internal? Existing public API should suffice for testability (reason we made it public in the first place)?